enterprise/server/ociregistry/ociregistry.go

```diff
@@ -410,17 +412,23 @@ func fetchBlobOrManifestFromCache(ctx context.Context, w http.ResponseWriter, bs
 	w.Header().Add(headerContentType, blobMetadata.GetContentType())
 	w.WriteHeader(http.StatusOK)
 
-	if writeBody {
-		blobRN := digest.NewCASResourceName(
-			blobCASDigest,
-			"",
-			repb.DigestFunction_SHA256,
-		)
-		blobRN.SetCompressor(repb.Compressor_ZSTD)
-		return cachetools.GetBlob(ctx, bsClient, blobRN, w)
-	} else {
+	if !writeBody {
 		return nil
 	}
+	blobRN := digest.NewCASResourceName(
+		blobCASDigest,
+		"",
+		repb.DigestFunction_SHA256,
+	)
+	blobRN.SetCompressor(repb.Compressor_ZSTD)
+	err = cachetools.GetBlob(ctx, bsClient, blobRN, w)
+	if err == nil {
```

Comment:
nit: error handling is usually written as `if err != nil`. Also would be good to return `stsatus.WrapError(err, "fetch blob from CAS")` so that it's easier to trace the error when looking at logs.

Reply:
If I use ioutil.Counter, I don't think I'll have to worry about capturing the non-error case.